### PR TITLE
Xcode Upgrade

### PR DESCRIPTION
**Pull request by jonallured**
@orta and I have been working on getting Folio to play nice with the new Xcode. I did a pod update and had to back off a couple, but the biggest change was to regenerate the snapshots for iOS 11.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">No CHANGELOG changes made</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by orta** [comment]
So this is "working" but there's fun animation edge cases 

screen shot 2017-09-26 at 5 47 51 pm

**Comment by orta** [comment]
Threw another an hour or two at this, it's *very hard* to make the custom navigation bar consistent and work across both screen sizes and view controllers.

**Comment by orta** [comment]
only 34 failing tests

**Comment by orta** [comment]
This needs to handle iPhone X too

**Comment by orta** [comment]
Built an abstraction that helps handle the top/bottom for most of the login/setup screens 

screen shot 2018-03-24 at 7 06 04 pm

**Comment by orta** [comment]
`/me nods`

screen shot 2018-03-24 at 10 09 52 pm

**Comment by orta** [comment]
OK, I'm happy with iPhone support, now back to iPad.

File: Classes/Controllers/Collections Controllers/Hosts/ARTabbedViewController.m [hunk 1]
```diff
@@ -230,20 +236,19 @@ - (UIView *)sortButton
...
-
+    UILabel *arrow = [[UILabel alloc] init];
     arrow.font = [UIFont serifFontWithSize:SORT_BUTTON_ARROW_FONT_SIZE];
     arrow.text = @"▼";
     arrow.backgroundColor = [UIColor artsyBackgroundColor];
     arrow.textColor = [UIColor artsyForegroundColor];
 
     [_sortButton insertSubview:arrow belowSubview:_sortButton.titleLabel];
+    [arrow alignTrailingEdgeWithView:_sortButton predicate:@"0"];
+    [arrow alignCenterYWithView:_sortButton predicate:@"0"];
```
Comment (orta) [inline Classes/Controllers/Collections Controllers/Hosts/ARTabbedViewController.m]: I fixed the animation when you scroll up on the sort picker text - I spent 5m thinking at how to do it with the current frames, then 15s writing it in auto-layout instead

File: Classes/Controllers/Navigation Controllers/ARAlbumEditNavigationController.m [hunk 2]
```diff
@@ -23,9 +21,6 @@ - (instancetype)initWithAlbum:(Album *)album
 
     _album = album;
 
-    _navDelegate = [[ARRootNavigationControllerDelegate alloc] init];
-    self.delegate = _navDelegate;
-
```
Comment (orta) [inline Classes/Controllers/Navigation Controllers/ARAlbumEditNavigationController.m]: Used to be used to set up the iPad toolbar, now removed.

File: Classes/Views/Custom Controls/ARSelectionToolbarView.m [hunk 3]
```diff
@@ -105,8 +105,13 @@ - (void)fillWidthButtons
...
             [button constrainLeadingSpaceToView:self.buttons[index - 1] predicate:nil];
         }
-        
-        [button alignTopEdgeWithView:self predicate:@"0"];
+
+        if (self.isAttachedToTop) {
+            [button alignBottomEdgeWithView:self predicate:@"0"];
+        } else if(self.isAttachedToBottom) {
+            [button alignTopEdgeWithView:self predicate:@"0"];
+        }
```
Comment (orta) [inline Classes/Views/Custom Controls/ARSelectionToolbarView.m]: basically so that safe areas can be respected

**Comment by orta** [comment]
OK, the Red is only because it was a deploy - so I'm merging

**Pull request merged**